src/audio/tdfb/tdfb.c

```diff
@@ -147,10 +176,116 @@ static int tdfb_init_coef(struct tdfb_comp_data *cd, int source_nch,
 		return -EINVAL;
 	}
 
-	coefp = ASSUME_ALIGNED(&config->data[0], 2);
+	if (config->num_angles > SOF_TDFB_MAX_ANGLES) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid num_angles %d",
+			    config->num_angles);
+		return -EINVAL;
+	}
+
+	if (config->beam_off_defined > 1) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid beam_off_defined %d",
+			    config->beam_off_defined);
+		return -EINVAL;
+	}
+
+	if (config->num_mic_locations > SOF_TDFB_MAX_MICROPHONES) {
+		comp_cl_err(&comp_tdfb, "tdfb_init_coef(), invalid num_mic_locations %d",
+			    config->num_mic_locations);
+		return -EINVAL;
+	}
+
+	/* In SOF v1.6 based beamformer topologies the legacy flag is set to indicate no
+	 * support for multiple angles, no mic locations, and no beam on/off switch.
+	 */
+	if (config->num_angles == 0 && config->num_mic_locations == 0) {
+		legacy = 1;
+		num_filters = config->num_filters;
+	} else {
+		num_filters = config->num_filters * (config->num_angles + config->beam_off_defined);
+	}
+
+	/* Skip filter coefficients */
+	coefp = tdfb_filter_seek(config, num_filters);
+
+	/* Get shortcuts to input and output configuration */
+	cd->input_channel_select = coefp;
+	coefp += config->num_filters;
+	cd->output_channel_mix = coefp;
+	coefp += config->num_filters;
+	cd->output_stream_mix = coefp;
+	coefp += config->num_filters;
+
+	if (legacy) {
```

Comment:
can we have a comment that explain what legacy means

Reply:
If we do keep this code, i think we should split the code paths into two separate functions and then have generic helpers.

Reply:
That's a good point, I added it because of released SOF versions with old format blob. I'd be happy to remove the support.